zipline/pipeline/term.py

```diff
@@ -157,79 +128,57 @@ def static_identity(cls, inputs, mask, window_length, domain, dtype):
         This is a classmethod so that it can be called from Term.__new__ to
         determine whether to produce a new instance.
         """
-        return (cls, inputs, mask, window_length, domain, dtype)
+        return (cls, domain, dtype)
 
     def _validate(self):
         """
         Assert that this term is well-formed.  This should be called exactly
         once, at the end of Term._init().
         """
-        if self.inputs is NotSpecified:
-            raise TermInputsNotSpecified(termname=type(self).__name__)
-        if self.window_length is NotSpecified:
-            raise WindowLengthNotSpecified(termname=type(self).__name__)
         if self.dtype is NotSpecified:
             raise DTypeNotSpecified(termname=type(self).__name__)
-        if self.mask is NotSpecified and not self.atomic:
-            # This isn't user error, this is a bug in our code.
-            raise AssertionError("{term} has no mask".format(term=self))
-
-        if self.window_length:
-            for child in self.inputs:
-                if not child.atomic:
-                    raise InputTermNotAtomic(parent=self, child=child)
 
-    @lazyval
-    def atomic(self):
-        """
-        Whether or not this term has dependencies.
+    @abstractproperty
+    def inputs(self):
+        raise NotImplementedError()
 
-        If term.atomic is truthy, it should have dataset and dtype attributes.
-        """
-        return len(self.inputs) == 0
+    @abstractproperty
+    def mask(self):
+        raise NotImplementedError
```

Comment:
Luckily it doesn't have an actual effect for us, but I agree, let's raise an instance.